core/src/Kokkos_Macros.hpp

```diff
@@ -652,4 +652,13 @@
 #define KOKKOS_THREAD_LOCAL __thread
 #endif
 
+#define KOKKOS_PROFILING_STRINGIFY(x) #x
+#define KOKKOS_PROFILING_TOSTRING(x) KOKKOS_PROFILING_STRINGIFY(x)
+#if defined(KOKKOS_PROFILING_ENABLE_SOURCE_INFO)
+#define KOKKOS_FILELINE_NAME(name)               \
+  (__FILE__ "KKSPLIT" KOKKOS_PROFILING_TOSTRING( \
+      __LINE__) "KKSPLIT" KOKKOS_PROFILING_TOSTRING(name))
```

Comment:
What's `"KKSPLIT"`?

Reply:
We were looking for a separator between the name and the file/line info. This turned into a surprisingly involved conversation. We started out with ":" as the separator, as in `file:line:name`. 

Then we realized that it could be hard to distinguish between names with colons in them with KOKKOS_PROFILING_ENABLE_SOURCE_INFO off, and file/line/name combinations with that on. So we decided on KKSPLIT, with it being a user error to have names including KKSPLIT. So names will be fed to tools like `fileKKSPLITlineKKSPLITname`, which they can string process into something more readable.

I'm open to suggestions on this, it was the least worst thing we could think of.

Reply:
I find it hard to read

Reply:
@dalg24: the idea is that you won't see that string in Kokkos Tools, tools will present that info however they want. Of course feel free to advocate for a different presentation, this is the rare topic I'm not too dogmatic about